test/pytest/test_xpedite/test_profiler/profile.py

```diff
@@ -1,180 +1,102 @@
+  reportProfiles = runXpediteReport(
+    runId, context, scenario, sampleFilePath=sampleFilePath, cpuInfoOverride=True
   )
-  reportProfiles = result.profiles
-  reportProfiles.transactionRepo = None # transactionRepo not stored in .xpd files
-  reportProfiles.cpuInfo.cpuId = baselineProfiles.cpuInfo.cpuId
-  if profileInfo.benchmarkPaths:
-    validateBenchmarks(len(profileInfo.benchmarkPaths), reportProfiles)
-    validateBenchmarks(len(profileInfo.benchmarkPaths), baselineProfiles)
-  
-  findDiff(reportProfiles.__dict__, baselineProfiles.__dict__)
-  assert reportProfiles == baselineProfiles
+  reportProfiles.transactionRepo = None
+  reportProfiles.cpuInfo.cpuId = scenario.cpuId
+  if scenario.benchmarkPaths:
+    benchmarkCount = len(scenario.benchmarkPaths)
+    validateBenchmarks(reportProfiles, benchmarkCount)
+  findDiff(reportProfiles.__dict__, scenario.baselineProfiles.__dict__)
+  assert reportProfiles == scenario.baselineProfiles
 
-def validateBenchmarks(benchmarks, profiles):
-  for profile in profiles:
-    assert len(profile.benchmarks) == benchmarks
-    for benchmark in profile.benchmarks.keys():
-      assert len(profile.benchmarks[benchmark].timelineCollection) == 1000
-
-def collectDataFiles(dataFolder):
+def validateBenchmarks(profiles, benchmarkCount):
   """
-  Collect data files from the test directory mapping xpedite run ids to data files for the run
-  """
-  runId = None
-  for fileName in os.listdir(dataFolder):
-    if fileName.endswith('.data'):
-      words = fileName.split('-')
-      runId = (words[2])
-      return runId
-
-def generateProfileInfo(binary, profileInfo, txnCount, threadCount, remote=None, workspace=None):
+  Validate the number of benchmark and number of timelines per benchmark
   """
-  Use Xpedite's profile information generator to generate a new profileInfo.py
-  file for an xpedite application and compare the new profile information to
-  baseline profile information
-  """
-  from xpedite.profiler.profileInfoGenerator import ProfileInfoGenerator
-  with TargetLauncher(binary, profileInfo, txnCount, threadCount, workspace, remote) as app:
-    probes = ProbeAdmin.loadProbes(app.xpediteApp)
-    profiler = os.path.join(os.path.dirname(TEST_DIR), 'scripts/bin/xpedite')
-    profileInfo.appInfo = os.path.join(app.tempDir, 'xpedite-appinfo.txt')
-    generator = ProfileInfoGenerator(
-      app.xpediteApp.executableName, profileInfo.appHost, os.path.basename(profileInfo.appInfo),
-      probes, profiler
-    )
-    generator.generate()
-  return generator.filePath
+  for profile in profiles:
+    assert len(profile.benchmarks) == benchmarkCount
+    for benchmark in profile.benchmarks.keys():
+      assert len(profile.benchmarks[benchmark].timelineCollection) == TXN_COUNT
 
-def buildNotebook(dataDir, binary, txnCount, threadCount, remote=None, profileInfoPath=None, runId=None, workspace=None):
+def buildNotebook(context, scenario): #pylint: disable=unused-argument
```

Comment:
why not remove unused parameters ?

Reply:
The arguments are removed the pylint disable was for: profileInfoPath=None
I will remove the pylint comment